docs/developers-guides/DevelopACustomImportPlugin.md

```diff
@@ -0,0 +1,96 @@
+
+For correct representation on the UI, there should be added specific parameters to the plugin:
+
+```java
+  @Override
+  public Map<String, ?> getPluginParams() {
+    Map<String, Object> params = new HashMap<>();
+    params.put(ALLOWED_COMMANDS, new ArrayList<>(pluginCommandMapping.get().keySet()));
+    params.put(COMMON_COMMANDS, new ArrayList<>(commonPluginCommandMapping.get().keySet()));
+    params.put(DESCRIPTION_KEY, DESCRIPTION);
+    params.put(METADATA, Map.of(IS_INTEGRATIONS_ALLOWED, false));
+    params.put("maxFileSize", MAX_FILE_SIZE);
+    params.put("acceptFileMimeTypes", "List of acceptFileMimeTypes"));
+    return params;
+  }
+
+  @Override
+  public IntegrationGroupEnum getIntegrationGroup() {
+    return IntegrationGroupEnum.IMPORT;
+  }
+```
+
+### Step 4: Define the Import Command
+Create a class for the 'import' command that must implement the [CommonPluginCommand](https://github.com/reportportal/plugin-api/blob/develop/src/main/java/com/epam/reportportal/extension/CommonPluginCommand.java) interface. This should:
+- Specify the command name in the `getName()` method, which must return "import".
+```java
+    @Override
+    public String getName() {
+      return "import";
+    }
+```
+
+- Handle the import logic in the `executeCommand(Map<String, Object> params)` method that includes file parsing and publishing launch reporting events.
+```java
+  @Override
+  public OperationCompletionRS executeCommand(Map<String, Object> params) {}
+```
+
+When working with a loaded file, it's necessary to trigger events in ReportPortal. For instance, if the report contains specific step information, you need to generate a StartTestItemRQ and publish a StartChildItemRqEvent:
+
+```java
+  private String startTestItem(ItemInfo itemInfo) {
+    StartTestItemRQ rq = buildStartItemRq(itemInfo);
+    eventPublisher.publishEvent(
+        new StartChildItemRqEvent(this, projectName, items.peek().getUuid(), rq));
+    return rq.getUuid();
+  }
+```
+
+The same approach used for other requests as well.
+
+### Step 5: Verify the plugin UI
+
+Provide the info, mentioned in [step 3](#step-3-implement-the-plugin-entrypoint). This will help the ReportPortal UI understand the plugin's capabilities and limitations.
+Once the step is completed, the plugin will be available for selection in the ReportPortal UI Launch import modal. No further actions required from the plugin UI to make the import process work. 
+
+#### Add custom UI extensions if needed
+
+The plugin template also provides a way to add custom extensions to the ReportPortal UI via plugin.
+This setup isn't required for import process work, but can be used to provide additional pages/sections in the ReportPortal UI via React components.
+For more information, see the [Plugin Developers Guide](./PluginDevelopersGuide.mdx). <!-- TODO: Update the link to the section with UI -->
+
+## Events to Manage
```

Comment:
Would it be helpful to add one more step related to plugin building process?
I suppose we can just refer another page in the docs, but it will help users to find the info easier.